yt/opensource-packages/ytsaurus/Dockerfile

```diff
@@ -0,0 +1,121 @@
+ARG SOURCE_ROOT="/ytsaurus"
+ARG BUILD_ROOT="/build"
+
+ARG PROTOC_VERSION="3.20.1"
+
+FROM ubuntu:focal as build-python-packages
+
+ARG SOURCE_ROOT
+ENV SOURCE_ROOT $SOURCE_ROOT
+
+ARG BUILD_ROOT
+ENV BUILD_ROOT $BUILD_ROOT
+
+RUN apt-get update && apt-get install -y software-properties-common
+RUN add-apt-repository ppa:deadsnakes/ppa
+
+RUN apt-get install -y --no-install-recommends \
+      curl \
+      unzip \
+      python3 \
+      python3-pip \
+      python3.11 \
+      python3.11-venv \
+    && rm -rf /var/lib/apt/lists/*
+
+ARG PROTOC_VERSION
+ENV PROTOC_VERSION $PROTOC_VERSION
+
+RUN curl -sL -o protoc.zip https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOC_VERSION}/protoc-${PROTOC_VERSION}-linux-x86_64.zip \
+    && unzip protoc.zip -d /usr/local \
+    && rm protoc.zip
+
+RUN mkdir -p ${SOURCE_ROOT}
+COPY ./yt/python ${SOURCE_ROOT}/yt/python
+COPY ./yt/yt/python ${SOURCE_ROOT}/yt/yt/python
+COPY ./certs/cacert.pem ${SOURCE_ROOT}/certs/cacert.pem
+
+RUN mkdir -p ${BUILD_ROOT}
+COPY ./artefacts/libyson_lib.so ${BUILD_ROOT}/yt/yt/python/yson_shared/libyson_lib.so
+COPY ./artefacts/libdriver_lib.so ${BUILD_ROOT}/yt/yt/python/driver/native_shared/libdriver_lib.so
+COPY ./artefacts/libdriver_rpc_lib.so ${BUILD_ROOT}/yt/yt/python/driver/rpc_shared/libdriver_rpc_lib.so
+
+RUN python3.11 -m venv package-env
+RUN . package-env/bin/activate; \
```

Comment:
I am not familiar with Dockerfile semantics. Do these changes of environment are visible in latter RUN commands?

Reply:
They aren't, so we have to do all of this within one RUN command.